diff --git a/src/hyperlight_guest/third_party/printf/printf.c b/src/hyperlight_guest/third_party/printf/printf.c
index 8a700ad..caf375a 100644
--- a/src/hyperlight_guest/third_party/printf/printf.c
+++ b/src/hyperlight_guest/third_party/printf/printf.c
@@ -149,9 +149,7 @@ static inline void _out_null(char character, void* buffer, size_t idx, size_t ma
 static inline void _out_char(char character, void* buffer, size_t idx, size_t maxlen)
 {
   (void)buffer; (void)idx; (void)maxlen;
-  if (character) {
     _putchar(character);
-  }
 }
 
 
